src/Symfony/Component/Utf8/CodePoints.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Utf8;
+
+use Symfony\Component\Utf8\Exception\InvalidArgumentException;
+
+final class CodePoints implements GenericStringInterface, \Countable
+{
+    use Utf8Trait;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function count(): int
+    {
+        return mb_strlen($this->string, 'UTF-8');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function length(): int
+    {
+        return mb_strlen($this->string, 'UTF-8');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function substr(int $start = 0, int $length = null): self
+    {
+        $result = clone $this;
+        $result->string = mb_substr($this->string, $start, $length, 'UTF-8');
+
+        return $result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getIterator(int $maxChunkLength = 1): \Generator
+    {
+        if (1 > $maxChunkLength) {
+            throw new InvalidArgumentException('The maximum length of each segment must be greater than zero.');
+        }
+
+        if ('' === $this->string) {
+            return;
+        }
+
+        if (65535 < $maxChunkLength) {
```

Comment:
As I said, I am not intelligent enough to deal with Yoda conditions. They require getting used to.

My point stands, this number is super magic. On top of all, this limit is not imposed on the `Bytes` implementation. Again a non uniform implementation of this generic code. A caller cannot even catch this particular issue separately because the very same exception is being reused.